sdk/eventhub/Azure.Messaging.EventHubs/tests/EventHubConsumer/EventHubConsumerLiveTests.cs

```diff
@@ -217,5 +224,1020 @@ await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsum
+        ///
+        [Test]
+        [TestCase("XYZ")]
+        [TestCase("-1")]
+        [TestCase("1000")]
+        [TestCase("-")]
+        public async Task ConsumerCannotReceiveFromInvalidPartition(string invalidPartition)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroup, invalidPartition, EventPosition.Latest))
+                    {
+                        Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero), Throws.InstanceOf<ArgumentOutOfRangeException>());
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        ///   Verifies that the <see cref="EventHubConsumer" /> is able to
+        ///   connect to the Event Hubs service and perform operations.
+        /// </summary>
+        ///
+        [Test]
+        [TestCase(1)]
+        [TestCase(3)]
+        [TestCase(6)]
+        [TestCase(10)]
+        [TestCase(20)]
+        [TestCase(50)]
+        public async Task ConsumerCannotReceiveMoreThanMaximumMessageCountMessages(int maximumMessageCount)
+        {
+            await using (var scope = await EventHubScope.CreateAsync(1))
+            {
+                var connectionString = TestEnvironment.BuildConnectionStringForEventHub(scope.EventHubName);
+
+                var eventBatch = Enumerable
+                    .Range(0, 20 * maximumMessageCount)
+                    .Select(index => new EventData(new byte[10]));
+
+                await using (var client = new EventHubClient(connectionString))
+                {
+                    var partition = (await client.GetPartitionIdsAsync()).First();
+
+                    await using (var producer = client.CreateProducer(new EventHubProducerOptions { PartitionId = partition }))
+                    await using (var consumer = client.CreateConsumer(EventHubConsumer.DefaultConsumerGroup, partition, EventPosition.Latest))
+                    {
+                        // Initiate an operation to force the consumer to connect and set its position at the
+                        // end of the event stream.
+
+                        Assert.That(async () => await consumer.ReceiveAsync(1, TimeSpan.Zero), Throws.Nothing);
+
+                        // Send the batch of events.
+
+                        await producer.SendAsync(eventBatch);
+
+                        // Receive and validate the events.
+
+                        var index = 0;
+
+                        while (index < eventBatch.Count())
+                        {
+                            var receivedEvents = await consumer.ReceiveAsync(maximumMessageCount, TimeSpan.FromSeconds(10));
```

Comment:
You probably want to `.ToList()` this, since an enumerable is technically not considered safe to enumerate multiple times.  It'll also make those `Count()` references more efficient.

Reply:
Done.